connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporterTest.java

```diff
@@ -78,4 +68,41 @@ public void testGetFutures() {
         reporter.awaitFutures(topicPartitions);
         assertTrue(reporter.futures.isEmpty());
     }
+
+    @Test
+    public void testReport() {
+        initializeReporter(true);
+        when(errorReporter.report(any())).thenReturn(CompletableFuture.completedFuture(null));
+        @SuppressWarnings("unchecked") ConsumerRecord<byte[], byte[]> consumerRecord = mock(ConsumerRecord.class);
+        when(record.originalRecord()).thenReturn(consumerRecord);
+        reporter.report(record, new Throwable());
+        verify(errorReporter).report(any());
+    }
+
+    @Test
+    public void testReportNoToleratedErrors() {
+        initializeReporter(false);
+        when(errorReporter.report(any())).thenReturn(CompletableFuture.completedFuture(null));
+        @SuppressWarnings("unchecked") ConsumerRecord<byte[], byte[]> consumerRecord = mock(ConsumerRecord.class);
+        when(record.originalRecord()).thenReturn(consumerRecord);
+        assertThrows(ConnectException.class, () -> reporter.report(record, new Throwable()));
+        verify(errorReporter).report(any());
+    }
```

Comment:
I like the new coverage from these test cases. They're very similar--can we consolidate?
```suggestion
    @Test
    public void testReport() {
        testSingleReport(true);
    }

    @Test
    public void testReportNoToleratedErrors() {
        testSingleReport(false);
    }

    private void testSingleReport(boolean errorsTolerated) {
        initializeReporter(errorsTolerated);
        when(errorReporter.report(any())).thenReturn(CompletableFuture.completedFuture(null));
        @SuppressWarnings("unchecked") ConsumerRecord<byte[], byte[]> consumerRecord = mock(ConsumerRecord.class);
        when(record.originalRecord()).thenReturn(consumerRecord);
        if (errorsTolerated) {
            reporter.report(record, new Throwable());
        } else {
            assertThrows(ConnectException.class, () -> reporter.report(record, new Throwable()));
        }
        verify(errorReporter).report(any());
    }
```